src/coreclr/jit/importer.cpp

```diff
@@ -4134,35 +4135,104 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
+                                               info.compCompHnd->asCorInfoType(hClass) != CORINFO_TYPE_PTR)
+                                                  ? 1
+                                                  : 0);
+                            break;
+                        case NI_System_Type_get_IsByRefLike:
+                            retNode = gtNewIconNode(
+                                (info.compCompHnd->getClassAttribs(hClass) & CORINFO_FLG_BYREF_LIKE) ? 1 : 0);
+                            break;
+                        default:
+                            NO_WAY("Intrinsic not supported in this path.");
+                    }
+                    impPopStack(); // drop CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPE call
+                }
+                break;
+            }
+
+            case NI_System_Type_GetTypeCode:
+            {
+                CORINFO_CLASS_HANDLE hClass = NO_CLASS_HANDLE;
+                if (gtIsTypeof(impStackTop().val, &hClass))
+                {
+                    int typeCode = -1;
+                    switch (info.compCompHnd->getTypeForPrimitiveValueClass(hClass))
+                    {
+                        case CORINFO_TYPE_BOOL:
+                            typeCode = 3;
+                            break;
+                        case CORINFO_TYPE_CHAR:
+                            typeCode = 4;
+                            break;
+                        case CORINFO_TYPE_BYTE:
+                            typeCode = 5;
+                            break;
+                        case CORINFO_TYPE_UBYTE:
+                            typeCode = 6;
+                            break;
+                        case CORINFO_TYPE_SHORT:
+                            typeCode = 7;
+                            break;
+                        case CORINFO_TYPE_USHORT:
+                            typeCode = 8;
+                            break;
+                        case CORINFO_TYPE_INT:
+                            typeCode = 9;
+                            break;
+                        case CORINFO_TYPE_UINT:
+                            typeCode = 10;
+                            break;
+                        case CORINFO_TYPE_LONG:
+                            typeCode = 11;
+                            break;
+                        case CORINFO_TYPE_ULONG:
+                            typeCode = 12;
+                            break;
+                        case CORINFO_TYPE_FLOAT:
+                            typeCode = 13;
+                            break;
+                        case CORINFO_TYPE_DOUBLE:
+                            typeCode = 14;
+                            break;
+                        case CORINFO_TYPE_NATIVEINT:
+                        case CORINFO_TYPE_NATIVEUINT:
+                        case CORINFO_TYPE_PTR:
+                            typeCode = 1;
+                            break;
+                        default:
```

Comment:
Can you move the whole part into a separate function? it's already too long and produces extremely weird code-formatting